### DISCUSSION TITLE
Track global position of control

### DISCUSSION THREAD

**Discussion by dotlogix** [AvaloniaUI/Avalonia#16864] [Q&A]
Hey there is there any way to get notified when the GLOBAL position of a control changes?

It is very easy to react to changes of the position in a in a container like grid or canvas, but you don't get any callback if the position changes because of a scroll event (ScrollViewer)
However I have a use case where I want to anchor sth to controls of a scrollable canvas.

But I just can't figure out a solution to this.
Anyone has an Idea how to do this?

**Comment by dotlogix** [comment]
![Screenshot 2024-06-13 184141](https://github.com/user-attachments/assets/74ac96fa-97ff-44d2-8868-9d4db30b83b1)
As you see here I want to draw a spline between two elements in a canvas there can be multiple connections and each connection basically has two endpoint's it has to track.
These connections are drawn in a background canvas so the nodes are always on top of any line. The issue now is that if I make the node canvas scrollable I don't get updates if the position of a node changes.

All I need is a way to track the connection points relative to the window and get updated when this position changes.

**Comment by maxkatz6** [comment]
> Hey there is there any way to get notified when the GLOBAL position of a control changes?

It's tricky, as global position can be changed outside of the layout system or even UI thread. Imagine render thread animations.
But if layout only is enough for you, EffectiveViewportChanged event should be what you need.
